### PR TITLE
Add default initializers for generic parameters

### PR DESCRIPTION
**Pull request by kirtchev-adacore**

### CODE REVIEW

File: src/generics.rst [hunk 1]
```diff
@@ -300,12 +330,18 @@ of an :t:`associated trait type`.
...
 
 :dp:`fls_ukarc98ceesz`
 :t:`[Generic argument]s` are subject to :t:`generic conformance`.
 
+.. rubric:: Dynamic Semantics
+
+:dp:`fls_eZBsyyEOogEn`
+The :t:`evaluation` of a :t:`constant argument` evaluates its
+:t:`block expression` or :t:`literal expression`.
+
```
Comment (Veykril) [inline src/generics.rst:344]: This happens at compile time though, not run time, is that still dynamic?

**Review by Veykril — DISMISSED** [review]

**Comment by Veykril** [comment]
didn't mean to approve, whoops

File: src/generics.rst [hunk 2]
```diff
@@ -80,6 +88,18 @@ A :t:`generic union` is a :t:`union` with :t:`[generic parameter]s`.
...
+A :t:`constant parameter initializer` is a :t:`construct` that provides the
+default :t:`value` of its related :t:`constant parameter`.
+
+:dp:`fls_p4yb8EAXlRU0`
+A :t:`constant parameter initializer` shall be a :t:`constant expression`.
+
+:dp:`fls_FnMfBo5zFXJd`
+The :t:`type` of a :t:`constant parameter initializer` of a
+:t:`constant parameter` and the :t:`type` of the :t:`type ascription` of
+the :t:`constant parameter` shall be :t:`unifiable`.
```
[the remark this answers is not in the document]
Comment (kirtchev-adacore) [reply src/generics.rst:101]: Ah, you are right, this just falls out of the general type inference / unification mechanism.

File: src/generics.rst (hunk 1 above)
Comment (kirtchev-adacore) [reply src/generics.rst:344]: Oh, correct, I keep forgetting that constant XXX is **constant**. :D

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
bors merge

**Pull request merged**